### ISSUE TITLE
(3.0.50) bad breakpoint highlighting in inner context in debugger

### ISSUE THREAD

**Issue by dalehenrich** [ericwinger/Jade#291]
from nested block frame you are presented with a `set breakpoint` menu item:
![image](https://user-images.githubusercontent.com/623951/48664435-5b50b080-ea53-11e8-8762-0860ab38435c.png)
In same debugger from home frame you are presented with a `clear breakpoint` menu item: 
![image](https://user-images.githubusercontent.com/623951/48664449-820ee700-ea53-11e8-8faa-6b191439efb3.png)

Ideally you would be presented with a unified set of debugger operations for every frame that has the same home method (presumably that's the issue - the home context is not being used by the debugger)

**Comment by LisaAlmarode** [comment]
When you have a debugger on a breakpoint within a block context, the debugger believes there is a breakpoint in the wrong place.  Possibly it's taking a source offset into the code within the block and applying that to the outer method.  

Note that "new" is not only highlighted as if it was a breakpoint (red box), the debugger is even willing to remove this non-breakpoint, which removes the breakpoint elsewhere in the method. Very confusing.


![ss1](https://user-images.githubusercontent.com/13142166/51499124-e87af300-1d7d-11e9-8458-e4299626375a.jpg)

The actual breakpoint is NOT highlighted.  It has the I-m-executing-here gray background.  This goes away if you switch windows or anything.  The menu does not allow you to remove the breakpoint.

![ss2](https://user-images.githubusercontent.com/13142166/51499131-f03a9780-1d7d-11e9-90a3-bf9e37e67a9e.jpg)

**Comment by LisaAlmarode** [comment]
This needs attention; still an issue and quite ugly.  I've created a new example that might make it easier to see and reproduce the problem.

As you can see, there are three breakpoints within the ensure block.  The debugger stops correctly.  But the highlights are on breakpoints in a different block and in the assert.

This is for any test class:

```
test_highlighBreaks
	| var i log |

	[
		var := WriteStream on: String new.
		log := String new.
		i := 20.
		var nextPutAll: '123'.
		log add: 'abc'.
		
	] ensure: 
	[
		log size < 5 ifTrue: [log add: 'def'].
		var contents size < 5 ifTrue: [log add: '456'].
		i = 20 ifTrue: [ i := 3].
	].

	self assert: i = 3
```


![image](https://user-images.githubusercontent.com/13142166/63299761-64450b80-c28b-11e9-9edd-d9eb4c470569.png)

**Comment by ericwinger** [comment]
Note to self - Look at step point highlighting while I'm in here.  @LisaAlmarode - Can you find any Jadeite issues relating to step point highlighting within blocks?

**Comment by ericwinger** [comment]
Believe the case Lisa outlined is fixed in 3.0.78

**Comment by LisaAlmarode** [comment]
in 3.0.78, I am happy to see the correct highlights when I open the debugger.

However...

If you highlight one of the tokens with a breakpoint (trying to remove it), you get the "Break at step point 22", NOT the "Clear break at step point 22", so you cannot remove the breakpoint in the debugger.

Worse: After you bring up the menu item to attempt to remove a breakpoint, which you cannot do, if you click to another frame on the debugger stack and then back to this method, the highlights for the breakpoints move to the wrong places. 

![image](https://user-images.githubusercontent.com/13142166/66680280-0696c500-ec25-11e9-98d0-007fd232b1f5.png)

**Comment by ericwinger** [comment]
Have a fix for both problems mentioned - incorrect "set" menu item and breakpoints moving in inner blocks. Fixed in 3.0.80
